src/fn/breakoutheaders.ts

```diff
@@ -0,0 +1,238 @@
+import {
+  length,
+  type AnyCircuitElement,
+  type PcbPlatedHole,
+  type PcbSilkscreenPath,
+} from "circuit-json"
+import { z } from "zod"
+import { rectpad } from "../helpers/rectpad"
+import { silkscreenRef, type SilkscreenRef } from "src/helpers/silkscreenRef"
+import { platedhole } from "src/helpers/platedhole"
+
+export const breakoutheaders_def = z.object({
+  fn: z.string(),
+  w: length.default("22.58mm"),
+  h: length.optional(),
+  left: length.optional().default(20),
+  right: length.optional().default(20),
+  top: length.optional().default(0),
+  bottom: length.optional().default(0),
+  p: length.default(length.parse("2.54mm")),
+  id: length.optional().default(length.parse("1mm")),
+  od: length.optional().default(length.parse("1.2mm")),
+})
+
+export type breakoutheaders_def = z.input<typeof breakoutheaders_def>
+
+const getHeight = (parameters: breakoutheaders_def): number => {
+  const params = breakoutheaders_def.parse(parameters)
+
+  // Calculate height based on the presence of left and right parameters
+  if (params.left && params.right) {
+    return Math.max(params.left, params.right) * params.p
+  }
+
+  if (params.left) {
+    return params.left * params.p
+  }
+
+  if (params.right) {
+    return params.right * params.p
+  }
+
+  return 51
+}
+type Point = { x: number; y: number }
+type Direction = "left" | "right" | "top" | "bottom"
+
+const getTriangleDir = (
```

Comment:
this doesn't return a direction, but it sounds like it returns a triangle direction.

`getTrianglePath` with named parameters would be better!